harness/determined/exec/harness.py

```diff
@@ -82,69 +83,75 @@ def maybe_load_checkpoint(
 
 def build_and_run_training_pipeline(env: det.EnvContext) -> None:
 
-    # Create the socket manager. The socket manager will connect to the master and read messages
-    # until it receives the rendezvous_info.
-    #
-    # TODO(ryan): Pull profiler hooks out of SocketManager and into their own layer.
-    with layers.SocketManager(env) as socket_mgr:
+    # Create a background thread that can be used to send events such as telemetry or
+    # profiling metrics to the master (asynchronously).
+    with create_event_trail_thread(noop=False) as event_trail:
```

Comment:
damn we need to find a better syntax for all these context managers.

Reply:
It doesn't need to be a context manager since this thread should run through the whole program. It should be a singleton instead.

Reply:
You have to clean up the threads, which is a synchronization point, which is well-suited to a context manager